log4j-core/src/main/java/org/apache/logging/log4j/core/layout/Rfc5424Layout.java

```diff
@@ -634,11 +640,142 @@ public static Rfc5424Layout createLayout(
             includes = null;
         }
 
-        return new Rfc5424Layout(config, facility, id, enterpriseNumber, includeMDC, newLine, escapeNL, mdcId,
+        return new Rfc5424Layout(config, facility, id, String.valueOf(enterpriseNumber), includeMDC, newLine, escapeNL, mdcId,
                 mdcPrefix, eventPrefix, appName, msgId, excludes, includes, required, StandardCharsets.UTF_8,
                 exceptionPattern, useTlsMessageFormat, loggerFields);
     }
 
+    public static class Rfc5424LayoutBuilder {
```

Comment:
Please take a look if this Builder meets your expectations

Reply:
Looks good to me.